packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -2,96 +2,214 @@ import * as React from 'react';
+      return deepmerge(
+        {
+          ...(styleProps.fixed && styles.fixed),
+          ...(styleProps.hideScrollbar && styles.hideScrollbar),
+          ...(styleProps.scrollableX && styles.scrollableX),
+          ...(styleProps.scrollableY && styles.scrollableY),
+        },
+        styles.scroller || {},
+      );
+    },
   },
-  /* Styles applied to the tablist element if `variant="scrollable"` and `visibleScrollbar={false}`. */
-  hideScrollbar: {
+)(({ styleProps }) => ({
+  position: 'relative',
+  display: 'inline-block',
+  flex: '1 1 auto',
+  whiteSpace: 'nowrap',
+  ...(styleProps.fixed && {
+    overflowX: 'hidden',
+    width: '100%',
+  }),
+  ...(styleProps.hideScrollbar && {
     // Hide dimensionless scrollbar on MacOS
     scrollbarWidth: 'none', // Firefox
     '&::-webkit-scrollbar': {
       display: 'none', // Safari + Chrome
     },
-  },
-  /* Styles applied to the ScrollButtonComponent component. */
-  scrollButtons: {},
-  /* Styles applied to the ScrollButtonComponent component if `allowScrollButtonsMobile={true}`. */
-  scrollButtonsHideMobile: {
-    [theme.breakpoints.down('sm')]: {
-      display: 'none',
+  }),
+  ...(styleProps.scrollableX && {
+    overflowX: 'auto',
+    overflowY: 'hidden',
+  }),
+  ...(styleProps.scrollableY && {
+    overflowY: 'auto',
+    overflowX: 'hidden',
+  }),
+}));
+
+const FlexContainer = experimentalStyled(
+  'div',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'FlexContainer',
+    overridesResolver: (props, styles) => {
+      const { styleProps } = props;
+      return deepmerge(
+        {
+          ...(styleProps.flexContainerVertical && styles.flexContainerVertical),
+          ...(styleProps.centered && styles.centered),
+        },
+        styles.flexContainer || {},
+      );
     },
   },
-  /* Styles applied to the TabIndicator component. */
-  indicator: {},
-});
+)(({ styleProps }) => ({
+  display: 'flex',
+  ...(styleProps.flexContainerVertical && {
+    flexDirection: 'column',
+  }),
+  ...(styleProps.centered && {
+    justifyContent: 'center',
+  }),
+}));
+
+const TabIndicator = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'MuiTabs',
+    slot: 'Indicator',
+    overridesResolver: (props, styles) => styles.indicator,
+  },
+)(({ styleProps, theme }) => ({
+  position: 'absolute',
+  height: 2,
+  bottom: 0,
+  width: '100%',
+  transition: theme.transitions.create(),
+  ...(styleProps.indicatorColor === 'primary' && {
+    backgroundColor: theme.palette.primary.main,
+  }),
+  ...(styleProps.indicatorColor === 'secondary' && {
+    backgroundColor: theme.palette.secondary.main,
+  }),
+  ...(styleProps.vertical && {
+    height: '100%',
+    width: 2,
+    right: 0,
+  }),
+}));
 
 const defaultIndicatorStyle = {};
 
-const Tabs = React.forwardRef(function Tabs(props, ref) {
+const Tabs = React.forwardRef(function Tabs(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiTabs' });
+  const theme = useTheme();
```

Comment:
I tried that before but when I run `yarn proptypes && yarn docs:api` this got removed.

diff --git a/docs/pages/api-docs/tabs.json b/docs/pages/api-docs/tabs.json
index f677fabb55..31c87c7e68 100644
--- a/docs/pages/api-docs/tabs.json
+++ b/docs/pages/api-docs/tabs.json
@@ -7,7 +7,6 @@
     "centered": { "type": { "name": "bool" } },
     "children": { "type": { "name": "node" } },
     "classes": { "type": { "name": "object" } },
-    "component": { "type": { "name": "elementType" } },
     "indicatorColor": {
       "type": { "name": "enum", "description": "'primary'<br>&#124;&nbsp;'secondary'" },
       "default": "'primary'"
diff --git a/docs/translations/api-docs/tabs/tabs.json b/docs/translations/api-docs/tabs/tabs.json
index 10c8875e19..683b0174e1 100644
--- a/docs/translations/api-docs/tabs/tabs.json
+++ b/docs/translations/api-docs/tabs/tabs.json
@@ -8,7 +8,6 @@
     "centered": "If <code>true</code>, the tabs are centered. This prop is intended for large views.",
     "children": "The content of the component.",
     "classes": "Override or extend the styles applied to the component. See <a href=\"#css\">CSS API</a> below for more details.",
-    "component": "The component used for the root node. Either a string to use a HTML element or a component.",
     "indicatorColor": "Determines the color of the indicator.",
     "onChange": "Callback fired when the value changes.<br><br><strong>Signature:</strong><br><code>function(event: object, value: any) =&gt; void</code><br><em>event:</em> The event source of the callback. <strong>Warning</strong>: This is a generic event not a change event.<br><em>value:</em> We default to the index of the child (number)",
     "orientation": "The component orientation (layout flow direction).",
diff --git a/packages/material-ui/src/Tabs/Tabs.js b/packages/material-ui/src/Tabs/Tabs.js
index 2d4b3bd6ae..2c64575214 100644
--- a/packages/material-ui/src/Tabs/Tabs.js
+++ b/packages/material-ui/src/Tabs/Tabs.js
@@ -710,15 +710,6 @@ Tabs.propTypes /* remove-proptypes */ = {
    * Override or extend the styles applied to the component.
    */
   classes: PropTypes.object,
-  /**
-   * @ignore
-   */
-  className: PropTypes.string,
-  /**
-   * The component used for the root node.
-   * Either a string to use a HTML element or a component.
-   */
-  component: PropTypes.elementType,
   /**
    * Determines the color of the indicator.
    * @default 'primary'

Reply:
Out of curiosity, why do we have isRTL and theme inside `useThemeProps`?

Reply:
I think I've added it so that we don't have to require the context again in case we need the theme, as it is already once read from context for resolving default props. For the `isRtl`, if I remember correctly, we said it make more sense to directly expose  it as a prop in the result, in case we decide to change the logic of how we read it from the theme. If this creates confusion, we may drop it...

Reply:
Ok, let's leave it as is at this moment, but leave this one open for visibility. We may come back to this later.